### PR TITLE
Updates for macOS Ventura on Apple Silicon with Qt6

### PR DESCRIPTION
**Pull request by sigmahour**
This contribution includes several changes I made while experimenting with input-leap on Apple Silicon. I addressed all the build warnings for both Apple and Intel Macs. The project now successfully works with qt6 installed via homebrew.

System Details:

    Operating System: macOS Ventura 13.4
    macOS SDK: 13.3
    QT Version: 6.5.0 (installed via homebrew)

The build does require a few steps in order to run without instantly crashing:

1. Create a symbolic link to the brotli common library from the homebrew installation. Here is the command I used:

```
ln -s /opt/homebrew/lib/libbrotlicommon.1.dylib /Applications/InputLeap.app/Contents/Frameworks/libbrotlicommon.1.dylib
```

2. Sign it:

```
codesign --force --deep --sign - /Applications/InputLeap.app
```

3. Enable in Accessibility:

Navigate to: `System Settings > Privacy & Security > Accessibility`
Remove and Re-add "InputLeap" using the "-" and "+" buttons at the bottom.

---

You can specify the version of Qt to build by setting the environment variable `QT_VERSION=5` or `QT_VERSION=6`. Default is set to version 6.

### CODE REVIEW

**Comment by shymega** [comment]
I know someone managed to get a 'tap' working with macOS on Apple Sillicon. Maybe that'll help figure out the issue?

Also, all CI seems to have failed, so unfortunately, we can't merge this. The CI to pay attention to is GitHub Actions - not Azure Pipelines.

With regards to the commits, it looks sane, but I'd like, before merging when CI passes, it's all squashed into one commit.

File: clean_build.sh
Comment (p12tic) [inline clean_build.sh:25]: This makes Barrier unusable on anything prior to 13.3

File: macos_environment.sh
Comment (p12tic) [inline macos_environment.sh:22]: This should be configurable

**Comment by vancluever** [comment]
> I know someone managed to get a 'tap' working with macOS on Apple Silicon. Maybe that'll help figure out the issue.

@shymega that was me :slightly_smiling_face: and the homebrew tap is here: https://github.com/vancluever/homebrew-input-leap

Also mentioned in the forums [here](https://github.com/input-leap/input-leap/discussions/1623) and [here](https://github.com/input-leap/input-leap/discussions/1517#discussioncomment-5298379).

The tap is currently pinned to my fork while I wait for #1640.

Cheers!

**Review by shymega — CHANGES_REQUESTED** [review]
This should, ideally, be in a separate PR.

**Comment by shymega** [comment]
You must have misunderstood me. I was referring specifically to the lines I referenced in my review.

**Comment by shymega** [comment]
Sorry, got busy with interviews.

I meant specifically the part I reviewed. There were (are?) some changes that didn't belong in this PR, but other parts did. The PR is still welcome, just split in sensible PRs.

**Pull request closed without merging**
